src/main/java/io/jenkins/plugins/mcp/server/extensions/BuildArtifactsExtension.java

```diff
@@ -0,0 +1,192 @@
+/*
+ *
+ * The MIT License
+ *
+ * Copyright (c) 2025, Derek Taubert.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package io.jenkins.plugins.mcp.server.extensions;
+
+import static io.jenkins.plugins.mcp.server.extensions.util.JenkinsUtil.getBuildByNumberOrLast;
+
+import hudson.Extension;
+import hudson.model.Run;
+import io.jenkins.plugins.mcp.server.McpServerExtension;
+import io.jenkins.plugins.mcp.server.annotation.Tool;
+import io.jenkins.plugins.mcp.server.annotation.ToolParam;
+import jakarta.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import jenkins.util.VirtualFile;
+import lombok.extern.slf4j.Slf4j;
+
+@Extension
+@Slf4j
+public class BuildArtifactsExtension implements McpServerExtension {
+
+    /*
+     * Suppression Rationale:
+     * - "rawtypes": Jenkins' Run.Artifact is a raw type in Jenkins' legacy API
+     * - "unchecked": Run.getArtifacts() returns raw List requiring unchecked conversion
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Tool(description = "Get the artifacts for a specific build or the last build of a Jenkins job")
```

Comment:
will need annotations = @Tool.Annotations(destructiveHint = false) 
as it's `true` per default